modules/markup/markdown/meta.go

```diff
@@ -88,7 +88,9 @@ func ExtractMetadataBytes(contents []byte, out interface{}) ([]byte, error) {
 		line := contents[start:end]
 		if isYAMLSeparator(line) {
 			front = contents[frontMatterStart:start]
-			body = contents[end+1:]
+			if end+1 < len(contents) {
```

Comment:
This isn't actually necessary.

End is guaranteed to be at most the last character of the slice and `slice[len(slice):]` never overflows

Reply:
> End is guaranteed to be at most the last character of the slice and slice[len(slice):] never overflows

Yup, however:

suppose `len(contents) == 1`, `end=len(contents)=1 (line 83)`, then `contents[end+1:]` is `contents[2:]`, which causes out-of-bound error.